test/blackbox-tests/test-cases/formatting/ocamlformat-not-installed.t

```diff
@@ -0,0 +1,20 @@
+Regression test that `dune build @fmt` exits cleanly if ocamlformat is not
+installed and no `.ocamlformat` file is found.
+
+See https://github.com/ocaml/dune/issues/10578 .
+
+  $ cat >dune-project <<EOF
+  > (lang dune 3.12)
+  > EOF
+
+We neeed an ocaml file and a dune rule to trigger the logic that would hit ocamlformat
+
+  $ touch dune foo.ml
+
+TODO: Check if ocamlformat is installed and figure out how to remove or mask it if so
+
+  $ ocamlformat
+
+  $ which ocamlformat
```

Comment:
I haven't figure out a nice clean way to ensure that ocamlformat is not installed yet. I'll keep thinking on it, but any suggestions would be welcome.

Reply:
Maybe you could try cleaning out `PATH` before calling Dune? It's admittedly much easier to add things in the `PATH` than to remove them.